spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -221,41 +206,24 @@ private Dataset<Row> buildManifestEntryDF(List<ManifestFile> manifests) {
   private List<ManifestFile> writeManifestsForUnpartitionedTable(
       Dataset<Row> manifestEntryDF, int numManifests) {
 
-    Broadcast<Table> tableBroadcast =
-        sparkContext().broadcast(SerializableTableWithSize.copyOf(table));
     StructType sparkType = (StructType) manifestEntryDF.schema().apply("data_file").dataType();
     Types.StructType combinedPartitionType = Partitioning.partitionType(table);
-
-    // we rely only on the target number of manifests for unpartitioned tables
-    // as we should not worry about having too much metadata per partition
-    long maxNumManifestEntries = Long.MAX_VALUE;
+    Types.StructType partitionType = spec.partitionType();
 
     return manifestEntryDF
         .repartition(numManifests)
```

Comment:
While I think this is fine, we do have adapative shuffle now, maybe it makes more sense to stop doing the size estimations ourselves and instead instead use a cloned session with AQE and Advisory Partition Size set? 

I don't think that's a perfect solution but it may be worth considering?

Reply:
I think using the advisory partition size is going to be tricky given that it is expressed in terms of shuffle block files. I feel our own estimation would still be much more precise.